File: pmcjs/scripts/iter2.pl

%ht = (
  'foo' => 31,
  'bar' => 12,
  'baz' => 99,
);

my $is_first = 1;
foreach my $k1 (keys %ht) {
  if ($is_first) {
    delete $ht{'baz'};
    $is_first = 0;
  }
  foreach my $k2 (keys %ht) {
    print "$k1 $ht{$k1} $k2 $ht{$k2}\n";
  }
}

File: pmcjs/scripts/iter2.js

use('Perl');

var m = {
  foo : 31,
  bar : 12,
  baz : 99
};

var is_first = true;
for (var i in m) {
  if (is_first) {
    delete m['baz'];
    is_first = false;
  }
  for (var j in m) {
    print("" + i + " " + m[i] + " " + j + " " + m[j]);
  }
}

